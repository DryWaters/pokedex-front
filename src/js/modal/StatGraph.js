import React from 'react';
import Stat from './Stat';

import '../../styles/modal/StatGraph.css'

class Stat extends React.Component {
  
  render() {
    return (
      <div className="statcontainer">
        <div><p>Stats</p></div>
        <div className="statgraph">
          {Object.keys(props.stats).map((stat, index) => {
            return <Stat key={index} statAmount={props.stats[stat]} />
          })}
        </div>
        <div className="statnames">
          <p className="statname">Hp</p>
          <p className="statname">Attack</p>
          <p className="statname">Defense</p>
          <p className="statname">Special Aattack</p>
          <p className="statname">Special Defense</p>
          <p className="statname">Speed</p>
        </div>
      </div>
    )
  }
};

export default StatGraph;