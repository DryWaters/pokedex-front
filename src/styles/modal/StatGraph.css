.statcontainer {
  width: 50%;
  display: flex;
  flex-direction: column;
}

.statgraph {
  height: 200px;
  border-bottom: 1px solid black;
  display: flex;
  justify-content: space-evenly;
}

.statnames {
  display: flex;
  justify-content: space-evenly
}

.statname {
  width: 10%;
  margin: 0, 5%, 0, 5%;
}

.stattitle {
  float: left;
}
