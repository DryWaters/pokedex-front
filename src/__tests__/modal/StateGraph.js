import React from 'react';
import { shallow } from 'enzyme';

import StatGraph from '../../js/modal/StatGraph';

const testStats = {
    "hp": 45,
    "attack": 49,
    "defense": 49,
    "special-attack": 65,
    "special-defense": 65,
    "speed": 45
  }

describe('StatGraph Component', () => {

  it('should render StatGraph component without crashing', () => {
    shallow(<StatGraph stats={testStats}/>);
  });

  it('should match StatGraph component snapshot', () => {
    const wrapper = shallow(<StatGraph stats={testStats}/>);
    expect(wrapper).toMatchSnapshot();
  });
  
});
