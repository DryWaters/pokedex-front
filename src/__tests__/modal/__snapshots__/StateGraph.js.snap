// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatGraph Component should match StatGraph component snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StatGraph
    stats={
        Object {
            "attack": 49,
            "defense": 49,
            "hp": 45,
            "special-attack": 65,
            "special-defense": 65,
            "speed": 45,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h3
                    className="stattitle"
          >
                    Stats
          </h3>
</div>,
        <div
          className="statgraph"
>
          <Stat
                    statAmount={45}
          />
          <Stat
                    statAmount={49}
          />
          <Stat
                    statAmount={49}
          />
          <Stat
                    statAmount={65}
          />
          <Stat
                    statAmount={65}
          />
          <Stat
                    statAmount={45}
          />
</div>,
        <div
          className="statnames"
>
          <p
                    className="statname"
          >
                    Hp
          </p>
          <p
                    className="statname"
          >
                    Attack
          </p>
          <p
                    className="statname"
          >
                    Defense
          </p>
          <p
                    className="statname"
          >
                    Special Aattack
          </p>
          <p
                    className="statname"
          >
                    Special Defense
          </p>
          <p
                    className="statname"
          >
                    Speed
          </p>
</div>,
      ],
      "className": "statcontainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3
            className="stattitle"
>
            Stats
</h3>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Stats",
            "className": "stattitle",
          },
          "ref": null,
          "rendered": "Stats",
          "type": "h3",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Stat
              statAmount={45}
/>,
            <Stat
              statAmount={49}
/>,
            <Stat
              statAmount={49}
/>,
            <Stat
              statAmount={65}
/>,
            <Stat
              statAmount={65}
/>,
            <Stat
              statAmount={45}
/>,
          ],
          "className": "statgraph",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "statAmount": 45,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "statAmount": 49,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "statAmount": 49,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "statAmount": 65,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "statAmount": 65,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "statAmount": 45,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="statname"
>
              Hp
</p>,
            <p
              className="statname"
>
              Attack
</p>,
            <p
              className="statname"
>
              Defense
</p>,
            <p
              className="statname"
>
              Special Aattack
</p>,
            <p
              className="statname"
>
              Special Defense
</p>,
            <p
              className="statname"
>
              Speed
</p>,
          ],
          "className": "statnames",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hp",
              "className": "statname",
            },
            "ref": null,
            "rendered": "Hp",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Attack",
              "className": "statname",
            },
            "ref": null,
            "rendered": "Attack",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Defense",
              "className": "statname",
            },
            "ref": null,
            "rendered": "Defense",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Special Aattack",
              "className": "statname",
            },
            "ref": null,
            "rendered": "Special Aattack",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Special Defense",
              "className": "statname",
            },
            "ref": null,
            "rendered": "Special Defense",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Speed",
              "className": "statname",
            },
            "ref": null,
            "rendered": "Speed",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h3
                        className="stattitle"
            >
                        Stats
            </h3>
</div>,
          <div
            className="statgraph"
>
            <Stat
                        statAmount={45}
            />
            <Stat
                        statAmount={49}
            />
            <Stat
                        statAmount={49}
            />
            <Stat
                        statAmount={65}
            />
            <Stat
                        statAmount={65}
            />
            <Stat
                        statAmount={45}
            />
</div>,
          <div
            className="statnames"
>
            <p
                        className="statname"
            >
                        Hp
            </p>
            <p
                        className="statname"
            >
                        Attack
            </p>
            <p
                        className="statname"
            >
                        Defense
            </p>
            <p
                        className="statname"
            >
                        Special Aattack
            </p>
            <p
                        className="statname"
            >
                        Special Defense
            </p>
            <p
                        className="statname"
            >
                        Speed
            </p>
</div>,
        ],
        "className": "statcontainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3
              className="stattitle"
>
              Stats
</h3>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Stats",
              "className": "stattitle",
            },
            "ref": null,
            "rendered": "Stats",
            "type": "h3",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Stat
                statAmount={45}
/>,
              <Stat
                statAmount={49}
/>,
              <Stat
                statAmount={49}
/>,
              <Stat
                statAmount={65}
/>,
              <Stat
                statAmount={65}
/>,
              <Stat
                statAmount={45}
/>,
            ],
            "className": "statgraph",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "statAmount": 45,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "statAmount": 49,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "statAmount": 49,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "statAmount": 65,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "statAmount": 65,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "statAmount": 45,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="statname"
>
                Hp
</p>,
              <p
                className="statname"
>
                Attack
</p>,
              <p
                className="statname"
>
                Defense
</p>,
              <p
                className="statname"
>
                Special Aattack
</p>,
              <p
                className="statname"
>
                Special Defense
</p>,
              <p
                className="statname"
>
                Speed
</p>,
            ],
            "className": "statnames",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hp",
                "className": "statname",
              },
              "ref": null,
              "rendered": "Hp",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Attack",
                "className": "statname",
              },
              "ref": null,
              "rendered": "Attack",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Defense",
                "className": "statname",
              },
              "ref": null,
              "rendered": "Defense",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Special Aattack",
                "className": "statname",
              },
              "ref": null,
              "rendered": "Special Aattack",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Special Defense",
                "className": "statname",
              },
              "ref": null,
              "rendered": "Special Defense",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Speed",
                "className": "statname",
              },
              "ref": null,
              "rendered": "Speed",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
